### PR TITLE
GH-924 stabilise PluginUITests

### LINKED ISSUE #924: running maven build yields failed tests related to the external libraries
```
02:32:18 Running org.eclipse.n4js.tests.externalPackages.RunExternalLibrariesPluginTest
02:32:27 Tests run: 8, Failures: 0, Errors: 4, Skipped: 0, Time elapsed: 10.028 sec <<< FAILURE! - in org.eclipse.n4js.tests.externalPackages.RunExternalLibrariesPluginTest
02:32:27 runClientWithAllOpenedWorkspaceProjects(org.eclipse.n4js.tests.externalPackages.RunExternalLibrariesPluginTest)  Time elapsed: 1.278 sec
02:32:27 runClientWithTwoDeletedWorkspaceProjects(org.eclipse.n4js.tests.externalPackages.RunExternalLibrariesPluginTest)  Time elapsed: 1.534 sec
02:32:27 runClientWithTwoClosedWorkspaceProjectsThenReopenThem(org.eclipse.n4js.tests.externalPackages.RunExternalLibrariesPluginTest)  Time elapsed: 0.896 sec  <<< ERROR!
02:32:27 org.eclipse.core.internal.resources.ResourceException: Save operation warnings.
02:32:27 	at org.eclipse.n4js.tests.externalPackages.RunExternalLibrariesPluginTest.runClientWithTwoClosedWorkspaceProjectsThenReopenThem(RunExternalLibrariesPluginTest.java:233)
02:32:27 
02:32:27 runClientWithAllDeletedWorkspaceProjects(org.eclipse.n4js.tests.externalPackages.RunExternalLibrariesPluginTest)  Time elapsed: 1.859 sec
02:32:27 runClientWithTwoDeletedWorkspaceProjectsThenReImportThem(org.eclipse.n4js.tests.externalPackages.RunExternalLibrariesPluginTest)  Time elapsed: 2.219 sec
02:32:27 runClientWithAllClosedWorkspaceProjects(org.eclipse.n4js.tests.externalPackages.RunExternalLibrariesPluginTest)  Time elapsed: 0.743 sec  <<< ERROR!
02:32:27 org.eclipse.core.internal.resources.ResourceException: Save operation warnings.
02:32:27 	at org.eclipse.n4js.tests.externalPackages.RunExternalLibrariesPluginTest.runClientWithAllClosedWorkspaceProjects(RunExternalLibrariesPluginTest.java:163)
02:32:27 
02:32:27 runClientWithTwoClosedWorkspaceProjectsWithTransitiveDependency(org.eclipse.n4js.tests.externalPackages.RunExternalLibrariesPluginTest)  Time elapsed: 0.699 sec  <<< ERROR!
02:32:27 org.eclipse.core.internal.resources.ResourceException: Save operation warnings.
02:32:27 	at org.eclipse.n4js.tests.externalPackages.RunExternalLibrariesPluginTest.runClientWithTwoClosedWorkspaceProjectsWithTransitiveDependency(RunExternalLibrariesPluginTest.java:183)
02:32:27 
02:32:27 runClientWithTwoClosedWorkspaceProjectsWithDirectDependency(org.eclipse.n4js.tests.externalPackages.RunExternalLibrariesPluginTest)  Time elapsed: 0.785 sec  <<< ERROR!
02:32:27 org.eclipse.core.internal.resources.ResourceException: Save operation warnings.
02:32:27 	at org.eclipse.n4js.tests.externalPackages.RunExternalLibrariesPluginTest.runClientWithTwoClosedWorkspaceProjectsWithDirectDependency(RunExternalLibrariesPluginTest.java:212)
02:32:27 
```

### PR DESCRIPTION
**Pull request by yoosiba**
resolves #924

### CODE REVIEW

File: testhelpers/org.eclipse.n4js.ui.tests.helper/src/org/eclipse/n4js/tests/builder/AbstractBuilderTest.java [hunk 1]
```diff
@@ -59,15 +58,10 @@
...
  * @author Sven Efftinge - Initial contribution and API
  */
-@SuppressWarnings("restriction")
 @RunWith(XtextRunner.class)
 @InjectWith(N4JSUiInjectorProvider.class)
 public abstract class AbstractBuilderTest {
 
-	static {
-		EclipseGracefulUIShutdownEnabler.enableOnce();
-	}
```
Comment (mmews-n4) [inline testhelpers/org.eclipse.n4js.ui.tests.helper/src/org/eclipse/n4js/tests/builder/AbstractBuilderTest.java]: Where to / Why did you remove the above?

File: testhelpers/org.eclipse.n4js.ui.tests.helper/src/org/eclipse/n4js/tests/util/EclipseUIUtils.java [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+		boolean wasInterrupted = false;
+
+		Stopwatch sw = Stopwatch.createStarted();
+		do {
+
+			editorIsActive = page.getActiveEditor() == editor;
+			if (!editorIsActive)
+
+				try {
+					if (runsInUI)
```
Comment (mmews-n4) [inline testhelpers/org.eclipse.n4js.ui.tests.helper/src/org/eclipse/n4js/tests/util/EclipseUIUtils.java]: dangling else here and a try inbetween. How about braces?

File: testhelpers/org.eclipse.n4js.ui.tests.helper/src/org/eclipse/n4js/tests/util/EclipseUIUtils.java [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+		final boolean runsInUI = UIUtils.runsInUIThread();
+		if (runsInUI)
+			LOGGER.warn("Waiting for editor runs in the UI thread which can lead to UI thread starvation.");
+
+		final long maxWait = 5000;
+		boolean editorIsActive = false;
+		boolean wasInterrupted = false;
+
+		Stopwatch sw = Stopwatch.createStarted();
```
Comment (mmews-n4) [inline testhelpers/org.eclipse.n4js.ui.tests.helper/src/org/eclipse/n4js/tests/util/EclipseUIUtils.java]: still needed?

File: testhelpers/org.eclipse.n4js.ui.tests.helper/src/org/eclipse/n4js/tests/util/ProjectTestsUtils.java [hunk 4]
```diff
@@ -360,42 +370,80 @@ public static void waitForUpdateEditorJob() {
...
 				wasInterrupted = true;
+				LOGGER.error("Waiting for jobs was interrupted after " + sw + ".", e);
 			}
-		} while (wasInterrupted && (end - start) < maxWait);
-		if (!foundJob) {
-			LOGGER.debug("No running nor waiting jobs found, maybe all have already finished.");
+		} while (sw.elapsed(TimeUnit.MILLISECONDS) < maxWait
+				&& foundJob == true
+				&& wasInterrupted == false
+				&& wasCancelled == false);
```
Comment (mmews-n4) [inline testhelpers/org.eclipse.n4js.ui.tests.helper/src/org/eclipse/n4js/tests/util/ProjectTestsUtils.java]: why check a boolean for equals? (booleanVar == false/true)

File: tests/org.eclipse.n4js.ui.tests/src/org/eclipse/n4js/tests/bugs/ProjectImporter.java [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+		this.rootFolder = rootFolder;
+	}
+
+	void importProjects() throws Exception {
+		for (final File file : rootFolder.listFiles()) {
+			if (file.exists() && file.isDirectory() && null != file.listFiles() && 0 < file.listFiles().length) {
+
+				if (!newHashSet(file.list()).contains("_project")) {
+					final String projectName = file.getName();
+					AbstractIDEBUG_Test.LOGGER.info("Project \'" + projectName
```
Comment (mmews-n4) [inline tests/org.eclipse.n4js.ui.tests/src/org/eclipse/n4js/tests/bugs/ProjectImporter.java]: What kind of logging concept is that? Does it mean that the ProjectImporter should only used by ``AbstractIDEBUG_Test``s?

File: tests/org.eclipse.n4js.ui.tests/src/org/eclipse/n4js/tests/bugs/WorkspaceInitializer.java [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+					// Re-enable after imports done.
+					toggleAutobuild(true);
+				}
+
+				return null;
+			}
+		});
+	}
+
+	private static void toggleAutobuild(final boolean enabled) {
```
Comment (mmews-n4) [inline tests/org.eclipse.n4js.ui.tests/src/org/eclipse/n4js/tests/bugs/WorkspaceInitializer.java]: There are AutobuildUtils that provide methods to toggle Autobuild.

File: testhelpers/org.eclipse.n4js.ui.tests.helper/src/org/eclipse/n4js/tests/builder/AbstractBuilderTest.java (hunk 1 above)
Comment (yoosiba) [reply testhelpers/org.eclipse.n4js.ui.tests.helper/src/org/eclipse/n4js/tests/builder/AbstractBuilderTest.java]: duplicate, there is one below: https://github.com/eclipse/n4js/pull/930/files/69acff612f1094f6118f9825a68c9edbe2700cf3#diff-f9fa75a8643c1a4ad55c64b462cce924R69

File: testhelpers/org.eclipse.n4js.ui.tests.helper/src/org/eclipse/n4js/tests/util/EclipseUIUtils.java (hunk 2 above)
Comment (yoosiba) [reply testhelpers/org.eclipse.n4js.ui.tests.helper/src/org/eclipse/n4js/tests/util/EclipseUIUtils.java]: ok...

File: testhelpers/org.eclipse.n4js.ui.tests.helper/src/org/eclipse/n4js/tests/util/EclipseUIUtils.java (hunk 3 above)
Comment (yoosiba) [reply testhelpers/org.eclipse.n4js.ui.tests.helper/src/org/eclipse/n4js/tests/util/EclipseUIUtils.java]: yes for condition and output messages

File: testhelpers/org.eclipse.n4js.ui.tests.helper/src/org/eclipse/n4js/tests/util/ProjectTestsUtils.java (hunk 4 above)
Comment (yoosiba) [reply testhelpers/org.eclipse.n4js.ui.tests.helper/src/org/eclipse/n4js/tests/util/ProjectTestsUtils.java]: somehow I find it more readable then 
```java
		} while (sw.elapsed(TimeUnit.MILLISECONDS) < maxWait
				&& foundJob
				&& !wasInterrupted
				&& !wasCancelled);
```

File: tests/org.eclipse.n4js.ui.tests/src/org/eclipse/n4js/tests/bugs/ProjectImporter.java (hunk 5 above)
Comment (yoosiba) [reply tests/org.eclipse.n4js.ui.tests/src/org/eclipse/n4js/tests/bugs/ProjectImporter.java]: I guess refactoring just kept the reference. I will change it
